src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -52,11 +60,18 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
     }
 
     public get name(): string {
-        return nonNullProp(this.server, 'name');
+        if (this._serverId) {
+            return this.connectionString.name;
```

Comment:
I don't understand why one uses name, but the other fullId? Can you give some real-life examples maybe?

Reply:
For subscription accounts, 
`connectionString.fullId = "test-configure-firewall-log.postgres.database.azure.com:5432"`
`this._serverId = "/subscriptions/70593b55-43fd-4d86-9326-b33ac00cfdf1/resourceGroups/myResourceGroup/providers/Microsoft.DBforPostgreSQL/servers/test-azure-postgresql"`

We are using `serverId` for setting credentials and configuring firewall. I felt that the serverId provides a little bit more precision when storing credentials and firewall, when compared to using just the `connectionString.fullId`.

Reply:
Okay I see what's happening here. A few suggestions:
- Please rename `_serverId` to `_azureId`

IMO, this makes it more obvious where the id is coming from and why it doesn't exist in some cases.

- Remove `ParsedPostgresConnectionString.name`
- Remove `PostgresServerTreeItem.name`
- Add `PostgresServerTreeItem.azureName`

`azureName` should be an optional param to the constructor, instead of something you try to parse from the connection string. Since `azureName` will have the type `string | undefined`, this will have better type-safety for Azure-specific scenarios. For example, it will force you to call `nonNullProp` inside of `deleteTreeItemImpl`, which would throw an error we can easily watch out for in telemetry. As another example, label would look like this:
```
public get label(): string {
    return this.azureName || this.connectionString.fullId;
}
```